initiatives/1749_Cloud-Native_Standards_Checklist.md

```diff
@@ -0,0 +1,362 @@
+
+## General best practices for containers:
+
+The containerization principles outlined herein include emerging definitions of agentic microservices (e.g., autonomous, signal-driven, reasoning-capable container systems), the recommendations in this general section are not specific to agentic use cases and apply to any containerized environment.
+
+General best practices include: **Security**, which covers minimizing attack surface and safeguarding container integrity; **Observability**, which focuses on collecting actionable metrics, logs, and traces to understand system behavior; and **Availability & Fault Tolerance**, which outlines strategies for maintaining service continuity and resilience under failure conditions.
+
+**Security**
+
+- [ ] Enforce the principle of least privilege for containers. Only grant the minimal permissions required for the container to operate, minimizing the attack surface.  This requires configuring user controls, network policies, security contexts of containers, and access control.   
+- [ ] Information hiding \- Avoid exposing unnecessary dependencies outside the container.   
+- [ ] Package only what is needed, use multi-stage builds to minimize image size, and avoid leaking build tools, credentials, or secrets.  
+- [ ] Use secure container images from official, trusted repositories, scanning images for vulnerabilities regularly. Sign and verify images to ensure the integrity and provenance of container images. Add OCI-compliant annotations to container images to document metadata such as source, version, authorship, scan status, and signature information.  
+- [ ] Follow secret management best practices. Never bake secrets into container images. Use Kubernetes Secrets or integrate with secret managers.  
+- [ ] Run containers as non-root users. Define a non-root user in the Dockerfile and configure the runtime to use it. This limits the blast radius in case of a security breach.  
+- [ ] Continuously monitor and update base images of the container to include the latest security patches and avoid known vulnerabilities. Use distroless images where possible.  
+- [ ] Log and monitor container activity. Monitor runtime behavior, resource usage, filesystem access, network activity, and system calls to detect anomalies or security incidents early.
+
+**Observability**
+
+- [ ] Use a standard observability stack (MELT). Consolidating metrics, logs, and traces to ensure viable explainability and achievable debugability of the system.   
+- [ ] Incorporate network observability by collecting flow logs for security, performance monitoring, and troubleshooting.  
+- [ ] Monitor resource-specific metrics  
+      - [ ] Disk Usage: Monitor disk usage on nodes and persistent volumes to prevent outages caused by storage exhaustion.  
```

Comment:
The preview shows no checkbox with `      `. Similarly, below and other blocks below. 
```suggestion
    - [ ] Disk Usage: Monitor disk usage on nodes and persistent volumes to prevent outages caused by storage exhaustion.  
```